### ISSUE TITLE
tf.losses.MeanSquaredError doesn't work for multi-dimensional labels and predictions

### ISSUE THREAD

**Issue by vivekkhimani** [tensorflow/tfjs#3755]
#### TensorFlow.js version
"@tensorflow/tfjs": "^2.0.1"
"@tensorflow/tfjs-react-native": "0.3.0"

#### Browser version
NA (iOS 10.0 on iPad 6th generation)

#### Describe the problem or feature request
I am trying to use **tf.losses.MeanSquaredError()** provided by tfjs to calculate the loss between 4-dimensional labels and predictions. We wouldn't ideally use 4-dimensional tensors while calculating the loss, but need it for a custom use case. The problem is that for any value of given label and prediction tensors [shape = (20, 32, 14, 14)], the loss is being returned as 1.0.

#### Code to reproduce the bug / link to feature request
JavaScript:
``` javascript
    import * as tf from '@tensorflow/tfjs';
  
    const js_a = Array(125440).fill(200);
    const a = tf.tensor4d(js_a, [20, 32, 14, 14]);
    const js_b = Array(125440).fill(1);
    const b = tf.tensor4d(js_b, [20, 32, 14, 14]);
    const mse_loss = tf.losses.meanSquaredError(a, b);
    mse_loss.print();
    // 1.0
```
Python:
```python
 import tensorflow as tf
 import numpy as np
 
 np_a = np.empty([20, 32, 14, 14])
 np_b = np.empty([20, 32, 14, 14])
 np_a.fill(200)
 np_b.fill(1)
 
 tf_a = tf.convert_to_tensor(np_a, np.float32)
 tf_b = tf.convert_to_tensor(np_b, np.float32)
 
 mse = tf.keras.losses.MeanSquaredError()
 print(mse(tf_a, tf_b).numpy())
 # 39600.758
```

GitHub issues for this repository are tracked in the [tfjs union repository](https://github.com/tensorflow/tfjs/issues).

Please file your issue there, following the guidance in [that issue template](https://github.com/tensorflow/tfjs/blob/master/ISSUE_TEMPLATE.md).

**Comment by kamalesh0406** [comment]
Even I faced this issue with the same version of tfjs.

**Comment by vivekkhimani** [comment]
UPDATE: The issue seems to be solved for "@tensorflow/tfjs": "^2.1.0". Closing it.
